### ISSUE TITLE
Really slow

### ISSUE THREAD

**Issue by jamestalmage** [meltingice/git-lfs-s3#12]
I've been able to get this working, but the client performs very slowly.
(This is true even without my HTTPS hack).

Testing against even just a few 2MB files, the client hangs for a very long time. The progress indicator shows the following for a long time:

`(0 of 1 files) 0 B / 1.74 MB`

It does eventually complete (sometimes).

Is this typical?

**Comment by meltingice** [comment]
Definitely not normal, but I don't believe it's an issue with git-lfs-s3. All it does is return a presigned URL for your git-lfs client to upload to. If the upload is taking a long time, then it's likely an issue with your internet connection, with AWS, or with git-lfs itself.

**Comment by jamestalmage** [comment]
I've tested against Githubs own service (it's fast). That seems to rule out internet connection and git-lfs.

I will try again later, but it seems.really unlikely the problem is S3 (I was experiencing the issue over the course of several hours)

**Comment by nigel-dev** [comment]
Just tested this and am running into the this same issue as @jamestalmage

**Comment by nterry** [comment]
Same here... It seems that if you press return (enter), it will continue...

**Comment by jamestalmage** [comment]
Yeah - I had that experience too. What is up with that??

**Comment by pranjaltech** [comment]
```
Git LFS: (0 of 1 files) 57.07 MB / 213.54 MB                                   Connection to github.com closed by remote host.
Git LFS: (2 of 1 files) 296.49 MB / 213.54 MB                                  
Put https://github-cloud.s3.amazonaws.com/alambic/media/***?actor_id=71014: read tcp 192.168.1.62:54463->54.231.8.177:443: read: network is down
error: failed to push some refs to 'git@github.com:pagalguy/******.git'
```

This happened with me. It uploaded way more data and then gave up.

**Comment by DALDEI** [comment]
Same here. gave up.

**Comment by nterry** [comment]
Yep, i have long since abandoned this piece of vaporware... Its unfortunate
that the developer has ignored these issues, as there is a lot of promise
for a project like this.

**Comment by meltingice** [comment]
I've been incredibly busy, but I would love to return to work on this project. I haven't forgotten about it.

## 

Ryan LeFevre (@meltingice)
Sr. Software Engineer
HODINKEE

On Sun, Jan 24, 2016 at 6:12 PM -0800, "Nicholas Terry" notifications@github.com wrote:

Yep, i have long since abandoned this piece of vaporware... Its unfortunate

that the developer has ignored these issues, as there is a lot of promise

for a project like this.

**Comment by kurtharriger** [comment]
I ran into this issue as well... however I find that if I restart git-lfs-s3 I can get it to upload.

git-lfs-s3
git push 
will hang
kill git-lfs-s3
git-lfs-s3
git will now start uploading the file but ultimately still fail
git push 
this time it will successfully verify and push succeeds

Here is the output from server. Hopefully this is info helps to narrow down where it gets stuck. 

https://gist.github.com/kurtharriger/ad9f7d3365eff74069c9

**Comment by luis-silva** [comment]
I'm facing similar issues where the 'git lfs push ....' hangs. After pressing return a few times the file eventually gets uploaded to S3 but not without a few errors:

Username for 'http://xxxx': 
Password for 'http://xxxx@xxxx':
Git LFS: (0 of 1 files) 0 B / 16.42 KB  
Git LFS: (0 of 1 files) 0 B / 16.42 KB
Git LFS: (0 of 1 files) 0 B / 16.42 KB
Git LFS: (0 of 1 files) 0 B / 16.42 KB                                                                                                                                                                                                                                         Git LFS: (0 of 1 files) 16.42 KB / 16.42 KB
Git LFS: (0 of 1 files) 16.42 KB / 16.42 KB
Git LFS: (1 of 1 files) 16.42 KB / 16.42 KB
Authorization error: http://xxxx/verify
Check that you have proper access to the repository
Authorization error: http://xxxx/verify
Check that you have proper access to the repository

git-lfs-s3[29680]: MYIP - - [26/May/2016:16:52:02 +0000] "POST /objects HTTP/1.1" 202 502 0.0372
git-lfs-s3[29680]: localhost - - [26/May/2016:16:52:02 UTC] "POST /objects HTTP/1.0" 202 502
git-lfs-s3[29680]: - -> /objects
git-lfs-s3[29680]: MYIP - - [26/May/2016:16:52:24 +0000] "POST /verify HTTP/1.1" 401 28 0.0006
git-lfs-s3[29680]: localhost - - [26/May/2016:16:52:24 UTC] "POST /verify HTTP/1.0" 401 28
git-lfs-s3[29680]: - -> /verify

Any update on this?
